addon/scheduler.ts

```diff
@@ -26,19 +33,51 @@ let _whenRoutePainted: Promise<any>;
 let _whenRoutePaintedScheduleFn: Function;
 let _whenRouteIdle: Promise<any>;
 let _whenRouteIdleScheduleFn: Function;
-let _activeScheduledTasks: number = 0;
 const CAPABILITIES: Capabilities = {
   requestAnimationFrameEnabled: typeof requestAnimationFrame === 'function',
   requestIdleCallbackEnabled: typeof requestIdleCallback === 'function',
+  performanceObserverEnabled: typeof PerformanceObserver === 'function',
 };
+export const DEFAULT_MARK_NAME: string = 'routeIdle';
+const DEFAULT_OPTIONS: SchedulerOptions = {
+  markName: DEFAULT_MARK_NAME,
+  emitMarks: true,
+};
+
 let _capabilities = CAPABILITIES;
+let schedulerOptions: SchedulerOptions = DEFAULT_OPTIONS;
 
 export const USE_REQUEST_IDLE_CALLBACK: boolean = true;
 export const SIMPLE_CALLBACK = (callback: Function) => callback();
 const IS_FASTBOOT = typeof (<any>window).FastBoot !== 'undefined';
+const waiter = buildWaiter('ember-app-scheduler-waiter');
+let _emitMarks: Function = emitMarks;
 
 reset();
 
+function installObserver() {
+  let observer = new PerformanceObserver(list => {
+    let entries = list.getEntriesByName(schedulerOptions.markName);
+
+    if (entries.length > 0) {
+      _whenRouteDidChange.resolve();
+    }
+  });
+  observer.observe({ entryTypes: ['mark'] });
+}
+
+export function emitMarks() {
```

Comment:
Lets make this method name singular, its going to emit a _single_ mark (each time you call it).